### PR TITLE
[cors] Add delegate function and change owner

### PR DESCRIPTION
**Pull request by pluma**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/expressjs/cors#configuring-cors-asynchronously
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

---

This seems to be an unintentional omission. The delegate has been supported as early as v1.0 and the typings claim to be compatible with v2.8 which still supports this.

Basically instead of providing the config synchronously, one can provide a delegate function which takes a request object and a callback and passes the configuration on a per request basis. This is probably a rare use case but certainly supported by the library.

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
Ping @mihhail-lapushkin

**Comment by mihhail-lapushkin** [comment]
I am totally out of TS community for years already. I don't know how to react to these requests. Is there a way I can resign the "ownership" of these modules?

**Comment by RyanCavanaugh** [comment]
@mihhail-lapushkin you can submit a PR to remove your name from the header

**Comment by pluma** [comment]
I can take over ownership of this one if you want. I don't know what other typings you contributed but maybe I can take care of some of those for you too?

**Comment by mihhail-lapushkin** [comment]
@pluma Thanks!
Here is the original PR for the modules that I've added: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/2815
Please, submit the PR to change the headers.

> I officially allow @pluma to take over the ownership of **cors** and **tea-merge** modules 😄

**Comment by pluma** [comment]
@mihhail-lapushkin I've added a commit to remove you from this module. I'm not familiar with tea-merge, though, and I think it's unrelated to this one. @RyanCavanaugh should I file a separate PR on @mihhail-lapushkin's behalf to remove him from tea-merge and leave it as unowned for the time being?

**Comment by RyanCavanaugh** [comment]
@pluma sounds fine by me. Please link to this PR from it since the next maintainer who sees it will otherwise reject it.

**Comment by pluma** [comment]
@RyanCavanaugh Can we get this PR merged then? I've removed him in this one and the linting is passing. I am also using the changes successfully as we speak.

@mihhail-lapushkin I've created PR #25223 to remove you from tea-merge as you requested.

**Comment by pluma** [comment]
btw, I'm confused that the type declaration declares a `namespace e`. Can this be safely removed?

**Comment by RyanCavanaugh** [comment]
@pluma the namespace is necessary in this case because the module exports a combination of a function and properties

**Comment by RyanCavanaugh** [comment]
🌟 🎈 🎉 🏆 🎂 ✨ ⭐️ 

Congratulations on your first DefinitelyTyped contribution! Thanks for helping out Mihhail as well.

🌟 🎈 🎉 🏆 🎂 ✨ ⭐️

**Comment by pluma** [comment]
Yay, thanks for the guidance :tada:
